iceoryx_hoofs/platform/mac/source/socket.cpp

```diff
@@ -17,6 +17,24 @@
 #include "iceoryx_hoofs/platform/socket.hpp"
 #include <unistd.h>
 
+#include <thread>
+
+static struct timeval getTimeoutOfSocket(int sockfd, int option_name)
+{
+    struct timeval tv;
+    socklen_t optionLength = sizeof(struct timeval);
+    if (getsockopt(sockfd, SOL_SOCKET, option_name, &tv, &optionLength) == -1)
+    {
+        return {0, 0};
+    }
+    return tv;
+}
+
+static void sleepFor(struct timeval& tv)
```

Comment:
Shouldn't this be marked with `noexcept`?

Reply:
I have no idea. This is in the platform headers and on windows/mac they actually throw exceptions. Here I wouldn't be that strict but we may need some kind of strategy how we handle such exceptions.

In linux/qnx the platform abstraction is in the end just a pure forwarding and there it is irrelevant.